python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -562,16 +541,23 @@ def _gaussianizeAndMeasure(self, measRecord: lsst.afw.table.SourceRecord,
         # scaling factors and sigmas.
         if measRecord.getFootprint().getArea() == 0:
             self._setFlag(measRecord, self.name, "no_pixel")
-            raise NoPixelError
+            self._setScalingAndSigmaFlags(measRecord, self.config.scalingFactors)
+            raise NoPixelError("No good pixels in footprint", 1)
 
         if (psf := exposure.getPsf()) is None:
             raise measBase.FatalAlgorithmError("No PSF in exposure")
 
         psfSigma = psf.computeShape(center).getTraceRadius()
         if not (psfSigma > 0):  # This captures NaN and negative values.
-            errorCollection = {str(scalingFactor): measBase.MeasurementError("PSF size could not be measured")
-                               for scalingFactor in self.config.scalingFactor}
-            raise GaapConvolutionError(errorCollection)
+            center = measRecord.getCentroid()
+            self.log.debug("Failed to solve for PSF matching kernel in GAaP for (%f, %f): %s",
+                           center.getX(), center.getY(), "GAaP Convolution Error")
+            self._setScalingAndSigmaFlags(
+                measRecord,
+                self.config.scalingFactors,
+                specificFlag="flag_gaussianization",
+            )
+            raise GaapConvolutionError("Failed to solve for PSF matching kernel", 1)
```

Comment:
I'd raise this as a plain `MeasurementError` and keep the error handling in `GaapConvolutionError` instead of trivially subclassing it from `MeasurementError`.

Reply:
I'm not sure what you mean.  I don't see the harm in having a slightly more specific error so that if you look at the debug log there's marginally more information (though there is still the message).  But I don't like having the error handling in the `Error` code because that's not what exceptions are supposed to do.

Reply:
Okay, if we want to keep error handling/error collating outside of `GaapConvolutionError`, then raising this specific type is fine.